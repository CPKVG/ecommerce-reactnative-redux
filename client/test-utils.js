// import { configureStore } from '@reduxjs/toolkit';
// import React from 'react';
// import { render as rtlRender } from '@testing-library/react-native'
// import { Provider } from 'react-redux';
// import { PersistGate } from 'redux-persist/integration/react';
// import { persistor } from './redux/createStore';
// import { rootReducer } from './redux/rootReducer';

// //custom render

// function render(
//     ui ,{
//         preloadedState,
//         store = configureStore({ reducer: rootReducer, preloadedState}),
//         ...renderOptions
//     } = {}
// ){
//   function Wrapper( {childern} ){
//       return <Provider store = {store}>
//           <PersistGate persistor={persistor}>
//             {childern}
//           </PersistGate>
//       </Provider>
//   }  
//   return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
// }

// //re-export everything
// export * from '@testing-library/react-native'
// //override render method
// export { render }


import React from "react"
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { rootReducer } from "./redux/rootReducer";

const INITIAL_STATE = { ...rootReducer }; // Or use value from root reducer
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
// const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    
    wrapper: ({ children }) => <
      Provider store={store}>
          {children}
      </Provider>,
    ...options
  });

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender as render, store };