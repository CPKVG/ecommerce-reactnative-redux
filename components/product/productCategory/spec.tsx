import React from "react"

import { NavigationContainer } from '@react-navigation/native';
import { render, store } from "../../../test-utils";
import ProductCategory from ".";
import { cleanup, fireEvent } from "@testing-library/react-native";
import { Image, Text, TouchableOpacity, View } from "react-native";
import { navigationRef } from "../../../RootNavigation";

import {Provider} from 'react-redux'
import thunk from 'redux-thunk';

import configureStore from 'redux-mock-store'
import Carousel from "react-native-snap-carousel";
import renderer from "react-test-renderer"
import ProductImage from "../productImage";


jest.useFakeTimers()

afterEach(cleanup)
const mockStore = configureStore([thunk])

describe('productCategory component', () => {
    // const store = mockStore({
    //     productSelector: {
    //         product: [''],
    //         loading:true,
    //         productData:'',
    //         productDataSearch:[undefined],
    //         productCategorySelect:"",
    //       },
    //   });

    // const component = render(
    //   <Provider store={store}>
    //     <ProductCategory />
    //   </Provider>,
    // )
    const component = (
        <NavigationContainer>
            <ProductCategory/>
        </NavigationContainer>
    )
  test('should render', () => { 
    expect(component).not.toBe(null)
  })

  test("if category image is selected, navigate search to product",()=>{

    const { queryByTestId  } = render(component) 
    const imageBtn:any = queryByTestId("carousel")

    fireEvent.press(imageBtn)
    
    expect(imageBtn).toBeTruthy
  })

it("renders the setctionList component",()=> {
    const data = [{"category":"data1"},{"category":"data2"}]
    const renderItem = ({item}:any)=>{
        return(
            <TouchableOpacity>
                <Text>
                {item.category}
                </Text>
                <Image
                source = {ProductImage(item.image)}
                />
            </TouchableOpacity>
        )
    }

    const tree = renderer.create(
        <View>
            <Carousel
              data= {data}
              renderItem = {renderItem}
              sliderWidth={400}
              itemWidth={125}
              loop={true}
              autoplay={true}
              autoplayDelay={25000}
              autoplayInterval={2500}
            />
        </View>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})


})

