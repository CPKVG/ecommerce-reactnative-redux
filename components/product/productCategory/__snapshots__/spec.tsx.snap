// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productCategory component renders the setctionList component 1`] = `
<View>
  <RCTScrollView
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={true}
    autoplayDelay={25000}
    autoplayInterval={2500}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 137.5,
          "paddingRight": 137.5,
        },
        Object {},
      ]
    }
    data={
      Array [
        Object {
          "category": "data2",
        },
        Object {
          "category": "data1",
        },
        Object {
          "category": "data2",
        },
        Object {
          "category": "data1",
        },
        Object {
          "category": "data2",
        },
        Object {
          "category": "data1",
        },
        Object {
          "category": "data2",
        },
        Object {
          "category": "data1",
        },
      ]
    }
    decelerationRate="fast"
    directionalLockEnabled={true}
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    initialNumToRender={11}
    invertStickyHeaders={false}
    inverted={false}
    itemWidth={125}
    keyExtractor={[Function]}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={false}
    loop={true}
    loopClonesPerSide={3}
    maxToRenderPerBatch={23}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderRelease={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={400}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 0,
        "width": 400,
      }
    }
    swipeThreshold={20}
    useScrollView={false}
    vertical={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={23}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data2
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data1
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data2
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data1
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data2
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data1
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data2
            </Text>
            <Image />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 125,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              data1
            </Text>
            <Image />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
